design/open/iceberg-branching.md

```diff
@@ -0,0 +1,80 @@
+# Iceberg support: branching & isolation
+
+## Problem Description
+
+Iceberg stores metadata files that represent a snapshot of a given Iceberg table.
+Those metadata files save the locataion of the data hard-coded.
+This is problematic with lakeFS, since we cannot work on the same files with different branches.
+
+For example when working on branch `main`, the metadata files will be written with the property `"location" : "s3a://example-repo/main/db/table1"`.
+This means the data is located in `"s3a://example-repo/main/db/table1"`.
+However, when branching out from main to `feature-branch`, the metadata files still point to the main branch.
+With this we lose the capability of isolation.
+
+
+## Goals
+
+Enable isolation with iceberg table. Create branch from existing one, and writing the data and metadata to the relevant branch.
+At first stage: support iceberg without being catalog agnostic - meaning work with only lakeFS catalog and not be compatible with other catalogs.
+At second stage: being catalog agnostic. Users will be able to configure both lakeFS catalog and other catalog together.
+- The catalogs that will be supported are Glue, Snoflake and Tabular.
+
+## Non Goals
+
+- Support merge operation.
+- Enable lakeFS operations using a dedicated SQL syntax.
```

Comment:
Will these be possible somehow?  Don't care about a detailed design, but _do_ prefer solutions that are steps along the way to having these -- particularly merges!

Reply:
Yes, enabling SQL syntax isn't related to own solution, and we do believe that our solution will fit to merge operation in the future.